src/main/java/LineTokenizer.java

```diff
@@ -0,0 +1,34 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+class LineTokenizer {
+
+    private final Map<String, List<Integer>> wordsChart;
```

Comment:
Do we really need this field? I can see that it’s only updated, but newer queried

Reply:
fixed